pkg/cmd/dictionary/entries/import/import.go

```diff
@@ -0,0 +1,225 @@
+	"github.com/algolia/algoliasearch-client-go/v3/algolia/search"
+	"github.com/spf13/cobra"
+
+	"github.com/algolia/cli/pkg/cmd/dictionary/shared"
+	"github.com/algolia/cli/pkg/cmdutil"
+	"github.com/algolia/cli/pkg/config"
+	"github.com/algolia/cli/pkg/iostreams"
+
+	"github.com/algolia/cli/pkg/prompt"
+	"github.com/algolia/cli/pkg/text"
+	"github.com/algolia/cli/pkg/utils"
+	"github.com/algolia/cli/pkg/validators"
+)
+
+type ImportOptions struct {
+	Config config.IConfig
+	IO     *iostreams.IOStreams
+
+	SearchClient func() (*search.Client, error)
+
+	DictionaryName    string
+	CreateIfNotExists bool
+	Wait              bool
+
+	File    string
+	Scanner *bufio.Scanner
+
+	ContinueOnError bool
+}
+
+// NewImportCmd creates and returns an import command for dictionary
+func NewImportCmd(f *cmdutil.Factory, runF func(*ImportOptions) error) *cobra.Command {
+	opts := &ImportOptions{
+		IO:           f.IOStreams,
+		Config:       f.Config,
+		SearchClient: f.SearchClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:       "import <dictionary> -F <file> [--wait] [--continue-on-errors]",
+		Args:      validators.ExactArgs(1),
+		ValidArgs: shared.DictionaryNames(),
+		Short:     "Import dictionary entries from a file to the specified index",
+		Long: heredoc.Doc(`
+			Import dictionary entries from a file to the specified index.
+			
+			The file must contains one single JSON object per line (newline delimited JSON objects - ndjson format: https://ndjson.org/).
+		`),
+		Example: heredoc.Doc(`
+			# Import entries from the "entries.ndjson" file to "stopwords" dictionary
+			$ algolia dictionary import stopwords -F entries.ndjson
+
+			# Import entries from the "entries.ndjson" file to "plurals" dictionary and continue importing entries even if some entries are invalid
+			$ algolia dictionary import plurals -F entries.ndjson --continue-on-errors
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.DictionaryName = args[0]
+
+			scanner, err := cmdutil.ScanFile(opts.File, opts.IO.In)
+			if err != nil {
+				return err
+			}
+			opts.Scanner = scanner
+
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return runImportCmd(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.File, "file", "F", "", "Read entries to import from `file` (use \"-\" to read from standard input)")
+	_ = cmd.MarkFlagRequired("file")
+
+	cmd.Flags().BoolVarP(&opts.Wait, "wait", "w", false, "Wait for the operation to complete before returning")
+	cmd.Flags().BoolVarP(&opts.ContinueOnError, "continue-on-error", "C", false, "Continue importing entries even if some entries are invalid.")
+
+	return cmd
+}
+
+func runImportCmd(opts *ImportOptions) error {
+	client, err := opts.SearchClient()
+	if err != nil {
+		return err
+	}
+
+	cs := opts.IO.ColorScheme()
+
+	var (
+		entries      []search.DictionaryEntry
+		currentLine  = 0
+		totalEntries = 0
+	)
+
+	// Scan the file
+	opts.IO.StartProgressIndicatorWithLabel(fmt.Sprintf("Reading words from %s", opts.File))
```

Comment:
Small nitpicking, you are using `words` here and `entries` later, might be confusing for the user.

Reply:
Good catch, thanks I forgot that one 👍